vowpalwabbit/csoaa.cc

```diff
@@ -470,6 +470,23 @@ void do_actual_learning(ldf& data, base_learner& base)
   if (start_K > 0)
     for (size_t k=1; k<K; k++)
       LabelDict::del_example_namespaces_from_example(*data.ec_seq[k], *data.ec_seq[0], (data.all->audit || data.all->hash_inv));
+
```

Comment:
I need ldf, so I cannot use `--oaa` (yes, VW is missing `--oaa_ldf`, but that's a minor problem). In `parse_args.cc` I check that the command line is `--csoaa_ldf=mc --loss_function=logisitic`, so the probabilities are well defined. See also my last comment in #566 .